server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/UserMappingRoutesTest.java

```diff
@@ -0,0 +1,229 @@
+        recipientRewriteTable = new MemoryRecipientRewriteTable();
+        DNSService dnsService = mock(DNSService.class);
+        DomainList domainList = new MemoryDomainList(dnsService);
+
+        domainList.addDomain(Domain.of("domain.tld"));
+        domainList.addDomain(Domain.of("abc"));
+        domainList.addDomain(Domain.of("xyz"));
+
+        recipientRewriteTable.setDomainList(domainList);
+
+        webAdminServer = WebAdminUtils.createWebAdminServer(new UserMappingRoutes(jsonTransformer, recipientRewriteTable))
+            .start();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath(UserMappingRoutes.BASE_PATH)
+            .log(LogDetail.METHOD)
+            .build();
+    }
+
+    @AfterEach
+    void stop() {
+        webAdminServer.destroy();
+    }
+
+    @Test
+    void getUserMappingsShouldReturnNotFoundByDefault() {
+        when()
+            .get()
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404)
+            .body("statusCode", is(HttpStatus.NOT_FOUND_404));
+    }
+
+    @Test
+    void getUserMappingsShouldBeEmptyWhenUserIsNotFound() {
+        when()
+            .get("randomuser@domain.tld")
+        .then()
+            .contentType(ContentType.JSON)
+            .statusCode(HttpStatus.OK_200)
+            .body(is("[]"));
+    }
+
+    @Test
+    void getUserMappingsShouldReturnCorrespondingMappingsFromUsername() throws Exception {
+        recipientRewriteTable.addAddressMapping(
+            MappingSource.fromUser(User.fromUsername("alice123@domain.tld")), "user123@domain.tld");
+        recipientRewriteTable.addAliasMapping(
+            MappingSource.fromUser(User.fromUsername("alice123@domain.tld")), "aliasuser123@domain.tld");
+        recipientRewriteTable.addGroupMapping(
+            MappingSource.fromUser(User.fromUsername("alice123@domain.tld")), "group123@domain.tld");
+
+        recipientRewriteTable.addAddressMapping(
+            MappingSource.fromUser(User.fromUsername("bob456@domain.tld")), "user456@domain.tld");
+        recipientRewriteTable.addAliasMapping(
+            MappingSource.fromUser(User.fromUsername("bob456@domain.tld")), "aliasuser456@domain.tld");
+        recipientRewriteTable.addGroupMapping(
+            MappingSource.fromUser(User.fromUsername("bob456@domain.tld")), "group456@domain.tld");
+
+        recipientRewriteTable.addAddressMapping(
+            MappingSource.fromUser(User.fromUsername("colin789@domain.tld")), "user789@domain.tld");
+        recipientRewriteTable.addAliasMapping(
+            MappingSource.fromUser(User.fromUsername("colin789@domain.tld")), "aliasuser789@domain.tld");
+        recipientRewriteTable.addGroupMapping(
+            MappingSource.fromUser(User.fromUsername("colin789@domain.tld")), "group789@domain.tld");
+
+        String jsonBody = when()
+                .get("alice123@domain.tld")
```

Comment:
Please extract your user name to class constant to improve readability and prevent typo mistakes.